### PR TITLE
KYLIN-4097: Throw exception when too many dict slice evictions in AppendTrieDictionary

### PR DESCRIPTION
**Pull request by lshmouse**
See: https://issues.apache.org/jira/browse/KYLIN-4097

### CODE REVIEW

File: core-common/src/main/java/org/apache/kylin/common/KylinConfigBase.java [hunk 1]
```diff
@@ -2219,4 +2219,8 @@ public int getStaleCubeThresholdInDays() {
     public int getStaleJobThresholdInDays() {
         return Integer.parseInt(getOptional("kylin.tool.health-check.stale-job-threshold-days", "30"));
     }
+
+    public int getDictionarySliceEvicationThreshold() {
```
Comment (nichunen) [inline core-common/src/main/java/org/apache/kylin/common/KylinConfigBase.java]: This config should be located in the domain of "DICTIONARY & SNAPSHOT"

**Comment by nichunen** [comment]
Hi @lshmouse  Would you please add a ut for this at AppendTrieDictionaryTest?

File: core-common/src/main/java/org/apache/kylin/common/KylinConfigBase.java (hunk 1 above)
Comment (lshmouse) [reply core-common/src/main/java/org/apache/kylin/common/KylinConfigBase.java]: done

**Comment by lshmouse** [comment]
@

> Hi @lshmouse Would you please add a ut for this at AppendTrieDictionaryTest?

OK

**Comment by lshmouse** [comment]
@nichunen  Added a UT for the case. Please help to review the PR. Thanks~

**Pull request merged**